streams/src/test/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdaterTest.java

```diff
@@ -163,21 +172,12 @@ public void shouldRestoreSingleActiveStatefulTask() throws Exception {
 
         stateUpdater.add(task);
 
-        final Set<StreamTask> expectedRestoredTasks = Collections.singleton(task);
-        final Set<StreamTask> restoredTasks = new HashSet<>();
-        waitForCondition(
-            () -> {
-                restoredTasks.addAll(stateUpdater.getRestoredActiveTasks(Duration.ofMillis(CALL_TIMEOUT)));
-                return restoredTasks.size() == expectedRestoredTasks.size();
-            },
-            VERIFICATION_TIMEOUT,
-            "Did not get any restored active task within the given timeout!"
-        );
-        assertTrue(restoredTasks.containsAll(expectedRestoredTasks));
-        assertEquals(expectedRestoredTasks.size(), restoredTasks.stream().filter(t -> t.state() == State.RESTORING).count());
+        verifyRestoredActiveTasks(task);
         assertTrue(stateUpdater.getAllTasks().isEmpty());
-        verify(changelogReader, atLeast(3)).restore(anyMap());
+        verify(changelogReader, times(1)).enforceRestoreActive();
+        verify(changelogReader, atLeast(1)).restore(anyMap());
```

Comment:
Why change from 3 to 1?

Reply:
I do not remember! Maybe a mistake. Reverted it to three.